File: solutions/go/hamming/hamming.go
package hamming

import "errors"

func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("different lengths")
	}

	var diffCount int
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			diffCount += 1
		}
	}
	return diffCount, nil
}

File: solutions/fortran/hamming/hamming.f90
module hamming
   implicit none
contains

   logical function compute(strand1, strand2, distance)
      character(*) :: strand1, strand2
      integer :: distance, i

      distance = 0
      compute = len(strand1) == len(strand2)
      if ( .not. compute ) return  ! different lengths

      do i = 1,len(strand1)
         if ( strand1(i:i) /= strand2(i:i) ) &
            distance = distance + 1
      end do
   end function

end module hamming
